packages/web/docs/src/components/masking-scrollview.tsx

```diff
@@ -0,0 +1,159 @@
+import { useEffect, useLayoutEffect, useReducer, useRef, useState } from 'react';
+
+export type MaskingScrollviewFade = {
+  y?: boolean;
+  x?: boolean;
+};
+
+export interface MaskingScrollviewProps {
+  fade: MaskingScrollviewFade;
+  children?: React.ReactNode;
+  className?: string;
+  outerClassName?: string;
+}
+
+export function MaskingScrollview({
+  fade,
+  children,
+  className,
+  outerClassName,
+  ...rest
+}: MaskingScrollviewProps) {
+  const scrollviewRef = useRef<HTMLDivElement | null>(null);
+  const { scrolledSides, shouldTransition } = useScrolledSides(scrollviewRef);
+
+  return (
+    <div
+      {...rest}
+      className={outerClassName}
+      style={{
+        // replace "mask" with "background" to debug it
+        maskImage: [
+          fade.x
+            ? `linear-gradient(to left, transparent, black 128px 25%, black 50%, transparent 50%),
+               linear-gradient(to right, transparent, black 128px 25%, black 50%, transparent 50%)`
+            : '',
+          fade.y
+            ? `linear-gradient(to bottom, transparent, black 128px 25%, black 50%, transparent 50%),
+               linear-gradient(to top, transparent, black 128px 25%, black 50%, transparent 50%)`
+            : '',
+        ]
+          .filter(Boolean)
+          .join(', '),
+        maskSize: [
+          fade.x ? 'calc(100% + 128px) 100%, calc(100% + 128px) 100%' : '',
+          fade.y ? '100% calc(100% + 128px), 100% calc(100% + 128px)' : '',
+        ]
+          .filter(Boolean)
+          .join(', '),
+        maskPosition: [
+          fade.x
+            ? `${scrolledSides.right ? '0px' : '-128px'} 0%, ${scrolledSides.left ? '0px' : '-128px'} 0%`
+            : '',
+          fade.y
+            ? `0% ${scrolledSides.top ? '-128px' : '0px'}, 0% ${scrolledSides.bottom ? '0px' : '-128px'}`
+            : '',
+        ]
+          .filter(Boolean)
+          .join(', '),
+        transition: shouldTransition
+          ? 'mask-position 0.5s ease, -webkit-mask-position 0.5s ease'
+          : '',
+      }}
+    >
```

Comment:
I would move this as a variable and memoize it

Reply:
I'd have probably try to argue and provide benchmarks of both options, **but** changing `fade` to `"x" | "y"` trivializes it to 3 ternaries.

🤔 If this was a public library component in a widely used design system, I'd probably move this logic to a className / Tailwind plugin, to allow `"max-sm:scrollview-fade-x sm:scrollview-fade-y"`.